tests/run-pass/async-fn.rs

```diff
@@ -38,4 +64,14 @@ fn main() {
     let waker = unsafe { Waker::from_raw(RawWaker::new(ptr::null(), &RAW_WAKER)) };
     let mut context = Context::from_waker(&waker);
     assert_eq!(unsafe { Pin::new_unchecked(&mut fut) }.poll(&mut context), Poll::Ready(31));
+
+    let mut fut = build_aggregate(1, 2, 3, 4);
+    let waker = unsafe { Waker::from_raw(RawWaker::new(ptr::null(), &RAW_WAKER)) };
+    let mut context = Context::from_waker(&waker);
+    assert_eq!(unsafe { Pin::new_unchecked(&mut fut) }.poll(&mut context), Poll::Ready(10));
+
+    let mut fut = includes_never(false, 4);
+    let waker = unsafe { Waker::from_raw(RawWaker::new(ptr::null(), &RAW_WAKER)) };
+    let mut context = Context::from_waker(&waker);
+    assert_eq!(unsafe { Pin::new_unchecked(&mut fut) }.poll(&mut context), Poll::Ready(16));
```

Comment:
Is it possible to move the common parts of these three "run an async fn" tests into a helper function?

I can also do that after landing, though.